projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/routing_policy/statement/PrependAsPath.java

```diff
@@ -48,20 +50,23 @@ public boolean equals(Object obj) {
   @Override
   public Result execute(Environment environment) {
     List<Integer> toPrepend = _expr.evaluate(environment);
+    List<SortedSet<Integer>> newAsPaths = new ArrayList<>(toPrepend.size());
+    // The ASes to prepend are in reverse order -- build up the prefix in the correct
+    // order.
+    for (Integer as : Lists.reverse(toPrepend)) {
+      newAsPaths.add(ImmutableSortedSet.of(as));
+    }
+
     BgpRoute.Builder bgpRouteBuilder = (BgpRoute.Builder) environment.getOutputRoute();
     List<SortedSet<Integer>> asPath = bgpRouteBuilder.getAsPath();
-    List<SortedSet<Integer>> iAsPath = null;
+    asPath.addAll(0, newAsPaths);
+
     if (environment.getWriteToIntermediateBgpAttributes()) {
       BgpRoute.Builder ir = environment.getIntermediateBgpAttributes();
-      iAsPath = ir.getAsPath();
-    }
-    for (Integer as : toPrepend) {
```

Comment:
I think the original and proposed implementations are wrong here. What I wanted to do was simply concatenate original list to the prepend list.